src/mpi/comm/comm_dup.c

```diff
@@ -149,8 +140,16 @@ int MPI_Comm_dup(MPI_Comm comm, MPI_Comm *newcomm)
             /* Validate comm_ptr */
             MPIR_Comm_valid_ptr( comm_ptr, mpi_errno, FALSE );
             if (mpi_errno) goto fn_fail;
+            /*If newcomm is NULL, creates and error to exit*/
+	    if(newcomm==NULL)
```

Comment:
The `MPIR_ERRTEST_ARGNULL` macro below should already be doing this. I think the case of `MPI_Comm_dup`, this check is already present.